### PR TITLE
CAD-1884: remove LiveView.

### PR DESCRIPTION
**Pull request by denisshevchenko**
Since `LiveView`-mode is officially deprecated and [RTView](https://github.com/input-output-hk/cardano-rt-view/) is in pre-release state, it's time to remove `LiveView` from the node.

Based on the short investigation of the community, we can say that most users already stopped using `LiveView` and use `gLiveView` script and/or Grafana instead.

### CODE REVIEW

**Review by deepfire — CHANGES_REQUESTED** [review]
Let's not merge this, until the community deprecation procedure is addressed.

**Comment by deepfire** [comment]
@denisshevchenko, what happens when you run a changed node against an old config file?  Will the node fail with an error?

If yes, then we should see what is the minimal change that allows it not to fail, but also complain about the deprecation to the user.

**Comment by denisshevchenko** [comment]
> what happens when you run the changed node against an old config file?

There's only _one_ change: no more `ViewMode`. So the user should remove that single value from node's config file.

File: cardano-node/cardano-node.cabal
Comment (Jimbo4350) [inline cardano-node/cardano-node.cabal:139]: Left over comment here

File: cardano-node/src/Cardano/Node/Types.hs
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Node/Types.hs:102]: Let's leave this `FromJSON` instance here and put an error message that explains `LiveView`/`SimpleView` has been deprecated and should be removed from the configuration file. Let devops and the wallet team know to expect this failure if their config yaml files haven't been updated accordingly. Let's replace `panic` with `fail` here as well.

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]
LGTM, two minor changes needed

File: cardano-node/src/Cardano/Node/Types.hs
Comment (denisshevchenko) [reply cardano-node/src/Cardano/Node/Types.hs:102]: Done. Now, if the config contains `ViewMode`, I see the message `AesonException "Error in $.ViewMode: ViewMode is deprecated, please remove it from your configuration file."`.

**Comment by denisshevchenko** [comment]
@deepfire Now, if the config contains `ViewMode`, I see the message `AesonException "Error in $.ViewMode: ViewMode is deprecated, please remove it from your configuration file."`.

**Comment by deepfire** [comment]
@denisshevchenko, I think it's worth running this through `weeder`: https://hackage.haskell.org/package/weeder

**Comment by deepfire** [comment]
So, if `weeder` is inaccessible due to GHC version, we could attack this from a different angle -- https://hackage.haskell.org/package/packunused

**Comment by mrBliss** [comment]
> So, if `weeder` is inaccessible due to GHC version, we could attack this from a different angle -- https://hackage.haskell.org/package/packunused

BTW, GHC 8.10 has the `-Wunused-packages` flag that does exactly what you need. You could try compiling `cardano-node` with GHC 8.10.2 and `-Werror=unused-packages`. At the time of writing this, I don't think the current master of `cardano-node` is compatible with GHC 8.10.2, though. According to https://github.com/input-output-hk/cardano-node/issues/1484#issuecomment-709435979, it will be as soon as #1958 is merged (you could temporarily rebase on that branch to try it out).

File: cardano-node/src/Cardano/Node/Types.hs
Comment (dcoutts) [inline cardano-node/src/Cardano/Node/Types.hs:105]: This does not make sense. This will cause a parse failure, not a warning.

File: cardano-node/src/Cardano/Node/Types.hs
Comment (dcoutts) [inline cardano-node/src/Cardano/Node/Types.hs:99]: I disagree. We should remove it. We can warn about deprecated config items without this.

File: cardano-node/src/Cardano/Node/Configuration/POM.hs
Comment (dcoutts) [inline cardano-node/src/Cardano/Node/Configuration/POM.hs:286]: We do not need this at all at this level. Indeed we can remove `pncViewMode` too.

If we really want to warn, we can do that when we read the json, but it cannot be a parse failure.

**Review by dcoutts — CHANGES_REQUESTED** [review]
I'm not convinced it is worth the effort to add a warning about the view mode field in the config that is no longer used.

If we do want to do that, we should do it comprehensively and warn about any field that is supplied but not used. I would not do that in this PR. Just remove it all.

File: cardano-node/src/Cardano/Node/Configuration/POM.hs
Comment (denisshevchenko) [reply cardano-node/src/Cardano/Node/Configuration/POM.hs:286]: Yes, @dcoutts, it's much simpler for me just remove it completely.

**Review by dcoutts — APPROVED** [review]
:tada:

File: cardano-node/cardano-node.cabal
Comment (intricate) [reply cardano-node/cardano-node.cabal:139]: @denisshevchenko

File: cardano-node/src/Cardano/Node/Configuration/Logging.hs
Comment (intricate) [inline cardano-node/src/Cardano/Node/Configuration/Logging.hs:39]: If we're no longer using the `UNIX` macro in this module, let's remove its definition and also the `CPP` language extension.

File: doc/logging-monitoring/liveview.md
Comment (intricate) [inline doc/logging-monitoring/liveview.md:3]: :+1:

**Review by intricate — CHANGES_REQUESTED** [review]

**Comment by mark-stopka** [comment]
Do I undertstand correctly that when merged, it will remove dependency on `vty` which require my Docker images to use an ugly workaround? 😍

**Comment by denisshevchenko** [comment]
@mark-stopka Yes, it will remove dependencies on vty and brick.

File: cardano-node/src/Cardano/Node/Configuration/Logging.hs
Comment (denisshevchenko) [reply cardano-node/src/Cardano/Node/Configuration/Logging.hs:39]: Done. But `CPP` is still required.

Comment (intricate) [reply cardano-node/src/Cardano/Node/Configuration/Logging.hs:39]: Right :+1:

**Review by deepfire — APPROVED** [review]

File: cardano-node/cardano-node.cabal
Comment (denisshevchenko) [reply cardano-node/cardano-node.cabal]: Fixed.

**Review by intricate — APPROVED** [review]

**Review by Jimbo4350 — APPROVED** [review]

**Pull request merged**
